spec/support/fake_tmpdir_helper.rb

```diff
@@ -0,0 +1,227 @@
+# = fake_tmpdir_helper.rb
+#
+# Spec::Support::FakeTmpdirHelper is a helper module that is used to simulate a
+# logical volume `/tmp` directory with limited disk space on the file system.
+#
+# To do this, it sets up a loopback device, which is just a zero'd out file
+# (defaults to `manageiq/tmp/fake_fs.img`), and mounts it.  `Dir.tmpdir` is
+# then overwritten to point to that newly mounted device, and it can
+# transparently be used to be tested against using file system commands like
+# normal (`df -P /fake_fs` for example).
+#
+# == Usage
+#
+# To use this lib, include it inside of your `describe` block:
+#
+#     describe 'Something' do
+#       include Spec::Support::FakeTmpdirHelper
+#     end
+#
+# Then use it by calling `with_fake_tmp` inside of the spec where you want to
+# have a "fake tmp dir".  It will do all of the setup and cleanup within the
+# block.
+#
+# Note:  This class is most likely **NOT** thread safe currently
```

Comment:
Will this be a problem for parallel tests? I mean probably not right now because there is only one spec using it ...